airflow/providers/hashicorp/common/vault_client.py

```diff
@@ -0,0 +1,392 @@
+        if kv_engine_version and kv_engine_version not in VALID_KV_VERSIONS:
+            raise VaultError(f"The version is not supported: {kv_engine_version}. "
+                             f"It should be one of {VALID_KV_VERSIONS}")
+        if auth_type not in VALID_AUTH_TYPES:
+            raise VaultError(f"The auth_type is not supported: {auth_type}. "
+                             f"It should be one of {VALID_AUTH_TYPES}")
+        if auth_type == "token" and not token:
+            raise VaultError("The 'token' authentication type requires 'token'")
+        if auth_type == "github" and not token:
+            raise VaultError("The 'github' authentication type requires 'token'")
+        if auth_type == "approle" and not role_id:
+            raise VaultError("The 'approle' authentication type requires 'role_id'")
+        if auth_type == "kubernetes":
+            if not kubernetes_role:
+                raise VaultError("The 'kubernetes' authentication type requires 'kubernetes_role'")
+            if not kubernetes_jwt_path:
+                raise VaultError("The 'kubernetes' authentication type requires 'kubernetes_jwt_path'")
+        if auth_type == "azure":
+            if not azure_resource:
+                raise VaultError("The 'azure' authentication type requires 'azure_resource'")
+            if not azure_tenant_id:
+                raise VaultError("The 'azure' authentication type requires 'azure_tenant_id'")
+        if auth_type == "radius":
+            if not radius_host:
+                raise VaultError("The 'radius' authentication type requires 'radius_host'")
+            if not radius_secret:
+                raise VaultError("The 'radius' authentication type requires 'radius_secret'")
+
+        self.kv_engine_version = kv_engine_version if kv_engine_version else 2
+        self.url = url
+        self.auth_type = auth_type
+        self.kwargs = kwargs
+        self.token = token
+        self.mount_point = mount_point if mount_point else\
+            self.get_default_mount_point(auth_type=self.auth_type,
+                                         kv_engine_version=self.kv_engine_version)
+        self.username = username
+        self.password = password
+        self.key_id = key_id
+        self.secret_id = secret_id
+        self.role_id = role_id
+        self.kubernetes_role = kubernetes_role
+        self.kubernetes_jwt_path = kubernetes_jwt_path
+        self.gcp_key_path = gcp_key_path
+        self.gcp_keyfile_dict = gcp_keyfile_dict
+        self.gcp_scopes = gcp_scopes
+        self.azure_tenant_id = azure_tenant_id
+        self.azure_resource = azure_resource
+        self.radius_host = radius_host
+        self.radius_secret = radius_secret
+        self.radius_port = radius_port
+
+    @staticmethod
+    def get_default_mount_point(auth_type: str, kv_engine_version: int = 2) -> str:
+        # pylint: disable=too-many-return-statements
+        """
+        Returns default mount point.
```

Comment:
Do we want this method, we can just pass if user provides one, if not, let's not pass anything which will uses the default

Reply:
Yep. Changed. That in the last fixup.